DQMOffline/L1Trigger/src/L1TStage2CaloLayer2Offline.cc

```diff
@@ -0,0 +1,596 @@
+      0.3);
+  h_resolutionJetPhi_HB_HE_ = ibooker.book1D("resolutionJetPhi_HB_HE",
+      "jet #phi resolution (HB+HE); (#phi_{jet}^{L1} - #phi_{jet}^{offline})/#phi_{jet}^{offline}; events", 120, -0.3,
+      0.3);
+
+  h_resolutionJetEta_ = ibooker.book1D("resolutionJetEta",
+      "jet #eta resolution  (HB); (L1 Jet #eta - Offline Jet #eta)/Offline Jet #eta; events", 120, -0.3, 0.3);
+
+  // jet turn-ons
+  ibooker.setCurrentFolder(efficiencyFolder_.c_str());
+  std::vector<float> jetBins(jetEfficiencyBins_.begin(), jetEfficiencyBins_.end());
+  int nBins = jetBins.size() - 1;
+  float* jetBinArray = &(jetBins[0]);
+
+  for (auto threshold : jetEfficiencyThresholds_) {
+    std::string str_threshold = std::to_string(int(threshold));
+    h_efficiencyJetEt_HB_pass_[threshold] = ibooker.book1D("efficiencyJetEt_HB_threshold_" + str_threshold + "_Num",
+        "jet efficiency (HB); Offline Jet E_{T} (GeV); events", nBins, jetBinArray);
+    h_efficiencyJetEt_HE_pass_[threshold] = ibooker.book1D("efficiencyJetEt_HE_threshold_" + str_threshold + "_Num",
+        "jet efficiency (HE); Offline Jet E_{T} (GeV); events", nBins, jetBinArray);
+    h_efficiencyJetEt_HF_pass_[threshold] = ibooker.book1D("efficiencyJetEt_HF_threshold_" + str_threshold + "_Num",
+        "jet efficiency (HF); Offline Jet E_{T} (GeV); events", nBins, jetBinArray);
+    h_efficiencyJetEt_HB_HE_pass_[threshold] = ibooker.book1D(
+        "efficiencyJetEt_HB_HE_threshold_" + str_threshold + "_Num",
+        "jet efficiency (HB+HE); Offline Jet E_{T} (GeV); events", nBins, jetBinArray);
+
+    h_efficiencyJetEt_HB_total_[threshold] = ibooker.book1D("efficiencyJetEt_HB_threshold_" + str_threshold + "_Den",
+        "jet efficiency (HB); Offline Jet E_{T} (GeV); events", nBins, jetBinArray);
+    h_efficiencyJetEt_HE_total_[threshold] = ibooker.book1D("efficiencyJetEt_HE_threshold_" + str_threshold + "_Den",
+        "jet efficiency (HE); Offline Jet E_{T} (GeV); events", nBins, jetBinArray);
+    h_efficiencyJetEt_HF_total_[threshold] = ibooker.book1D("efficiencyJetEt_HF_threshold_" + str_threshold + "_Den",
+        "jet efficiency (HF); Offline Jet E_{T} (GeV); events", nBins, jetBinArray);
+    h_efficiencyJetEt_HB_HE_total_[threshold] = ibooker.book1D(
+        "efficiencyJetEt_HB_HE_threshold_" + str_threshold + "_Den",
+        "jet efficiency (HB+HE); Offline Jet E_{T} (GeV); events", nBins, jetBinArray);
+  }
+
+  ibooker.cd();
+}
+
+//
+// -------------------------------------- functions --------------------------------------------
+//
+double L1TStage2CaloLayer2Offline::Distance(const reco::Candidate & c1, const reco::Candidate & c2)
+{
+  return deltaR(c1, c2);
+}
+
+double L1TStage2CaloLayer2Offline::DistancePhi(const reco::Candidate & c1, const reco::Candidate & c2)
+{
+  return deltaPhi(c1.p4().phi(), c2.p4().phi());
+}
+
+// This always returns only a positive deltaPhi
+double L1TStage2CaloLayer2Offline::calcDeltaPhi(double phi1, double phi2)
```

Comment:
this function should be replaced by std::abs() of DataFormats/Math/include/deltaPhi.h